crates/autopilot/src/solvable_orders.rs

```diff
@@ -586,14 +605,24 @@ async fn filter_unsupported_tokens(
     Ok(orders)
 }
 
+fn filter_limit_orders_with_insufficient_sell_amount(mut orders: Vec<Order>) -> Vec<Order> {
+    orders.retain(|order| match &order.metadata.class {
+        OrderClass::Limit(limit) => order.data.sell_amount > limit.surplus_fee,
+        _ => true,
+    });
+    orders
+}
```

Comment:
Looks like `filter_mispriced_limit_orders()` relies on this function being executed before hand (which might be changed by someone for whatever reason). Why not use `checked_sub()` in `filter_mispriced_limit_orders()` to avoid that issue altogether?

Reply:
Good point.

So my main motivation is that we want these two functions to filter different things (i.e. if we remove the `filter_mispriced_limit_orders` in favour of #765, we still keep this in).

I agree that `filter_mispriced_limit_orders` shouldn't rely on this - so I will add a `checked_sub` there as well.

Reply:
Added.